### PR TITLE
Layer3 acceptance tests dont use floating-network for subnets

### PR DESCRIPTION
**Pull request by dbaumgarten**
For #272 

Layer3 acceptance tests now use the network provided via OS_NETWORK_NAME to create their subnets instead of OS_EXTGW_ID.

No source/documentation to link to.

### CODE REVIEW

File: acceptance/openstack/networking/v2/extensions/layer3/layer3.go [hunk 1]
```diff
@@ -246,3 +248,24 @@ func WaitForRouterInterfaceToDetach(client *gophercloud.ServiceClient, routerInt
 		return false, nil
 	})
 }
+
+func GetNetworkIdByName(client *gophercloud.ServiceClient, name string)(string,error){
```
Comment (jtopjian) [inline acceptance/openstack/networking/v2/extensions/layer3/layer3.go]: Isn't is possible to use the existing `IDFromName` function in the `networks` package?

**Comment by jrperritt** [comment]
@dbaumgarten You've committed a file that has 16,000+ lines

**Comment by dbaumgarten** [comment]
Using IDFromName from the networks package is definetly the right thing to do. I just wasn't aware of its existence.
I also just noticed this strange (and huge) file. It seems like i messed  up when using git...
I will fix this right now.

**Comment by jtopjian** [comment]
@dbaumgarten Short of the above, the test looks good to me. Thanks for fixing it up.

As an aside, there are some existing scripts [here](https://github.com/gophercloud/gophercloud/tree/master/script/acceptancetest_environments) to build out devstack environments for acceptance tests. They handle setting the required environment variables like [`OS_NETWORK_ID`](https://github.com/gophercloud/gophercloud/blob/master/script/acceptancetest_environments/keystonev3-lbaasv2.sh#L198) which would be the equivalent of what one would set `OS_NETWORK_NAME` to, so it's possible to just nix the whole `OS_NETWORK_NAME` part.

As a second aside, [these](https://github.com/jtopjian/terraform-devstack) are more up-to-date scripts that I've been using. I should either replace the existing scripts in this repo to add a link pointing to them. [packstack-standard](https://github.com/jtopjian/terraform-devstack/tree/master/packstack-standard) has been my daily build environment for the past few months. Devstack started to get too hairy to maintain.

**Comment by dbaumgarten** [comment]
@jtopjian Thanks for the hint. Using a standadised testing environment seems like a good idea.

Desprite the test-environment-scripts setting OS_NETWORK_ID, there is no trace of this variable in AcceptanceTestChoicesFromEnv() and it isn't documented in the acceptance readme.

Should we add OS_NETWORK_ID to AcceptanceTestChoicesFromEnv() and/or the readme?

**Comment by jtopjian** [comment]
whoops! Looks like I got Terraform and Gophercloud mixed up. `OS_NETWORK_ID` is used in Terraform.

In that case, `OS_NETWORK_NAME` is totally fine. For my own sanity I might update Gophercloud to use `OS_NETWORK_ID` in the future. If you'd like to do that, go for it :)

**Pull request merged**